colibre-zooms/scripts/birth_density_distribution.py

```diff
@@ -122,17 +123,42 @@ def critical_density_DVS2012(
         "$z > 3$": birth_densities[birth_redshifts > 3],
     }
 
-    # Compute the critical density from DV&S2012
-    SNII_heating_temperature = float(
-        snapshot.metadata.parameters["COLIBREFeedback:SNII_delta_T_K"].decode("utf-8")
-    )  # in K
-    N_ngb_target = snapshot.metadata.hydro_scheme["Kernel target N_ngb"][0]
-    X_H = snapshot.metadata.hydro_scheme["Hydrogen mass fraction"][0]
-    M_gas = snapshot.metadata.initial_mass_table.gas.to("Msun")  # in Solar Masses
+    # Compute the critical density from DV&S201
+    try:
+        SNII_heating_temperature_min = float(
+            snapshot.metadata.parameters["COLIBREFeedback:SNII_delta_T_K_min"].decode(
+                "utf-8"
+            )
+        )  # in K
+        SNII_heating_temperature_max = float(
+            snapshot.metadata.parameters["COLIBREFeedback:SNII_delta_T_K_max"].decode(
+                "utf-8"
+            )
+        )  # in K
+        N_ngb_target = snapshot.metadata.hydro_scheme["Kernel target N_ngb"][0]
+        X_H = snapshot.metadata.hydro_scheme["Hydrogen mass fraction"][0]
+        M_gas = snapshot.metadata.initial_mass_table.gas.to("Msun")  # in Solar Masses
+
+        # Critical density corresponding to minimum heating temperature
+        n_crit_min = critical_density_DVS2012(
+            T_K=SNII_heating_temperature_min,
+            M_gas=M_gas.value,
+            N_ngb=N_ngb_target,
+            X_H=X_H,
+        )
+        # Critical density corresponding to maximum heating temperature
+        n_crit_max = critical_density_DVS2012(
+            T_K=SNII_heating_temperature_max,
+            M_gas=M_gas.value,
+            N_ngb=N_ngb_target,
+            X_H=X_H,
+        )
 
-    n_crit = critical_density_DVS2012(
-        T_K=SNII_heating_temperature, M_gas=M_gas.value, N_ngb=N_ngb_target, X_H=X_H
-    )
+    # Cannot find argument(s)
+    except KeyError:
+        print(traceback.format_exc())
```

Comment:
Is this still useful or is this a leftover from debugging?

Reply:
I decided to keep it purposely. I think it is useful to have so that whenever there is a key error, we will see it immediately in the output.

If you think that these lines make the code less readable or/and that we should not have any explicit traceback references and output (unless the pipeline is run in the debug mode), let me know and I will remove the lines (in this and in the other files)

Reply:
That's fine with me then. Let's keep it.